hoomd/hpmc/external/field.py

```diff
@@ -1,243 +1,223 @@
-                cls = _hpmc.ExternalFieldLatticeSpheropolygon
-            elif isinstance(mc, integrate.faceted_ellipsoid):
-                cls = _hpmc.ExternalFieldLatticeFacetedEllipsoid
-            elif isinstance(mc, integrate.polyhedron):
-                cls = _hpmc.ExternalFieldLatticePolyhedron
-            elif isinstance(mc, integrate.sphinx):
-                cls = _hpmc.ExternalFieldLatticeSphinx
-            elif isinstance(mc, integrate.sphere_union):
-                cls = _hpmc.ExternalFieldLatticeSphereUnion
-            elif isinstance(mc, integrate.faceted_ellipsoid_union):
-                cls = _hpmc.ExternalFieldlatticeFacetedEllipsoidUnion
-            elif isinstance(mc, integrate.convex_spheropolyhedron_union):
-                cls = _hpmc.ExternalFieldLatticeConvexPolyhedronUnion
-            else:
-                hoomd.context.current.device.cpp_msg.error(
-                    "compute.position_lattice_field: Unsupported integrator.\n")
-                raise RuntimeError(
-                    "Error initializing compute.position_lattice_field")
-        else:
-            hoomd.context.current.device.cpp_msg.error("GPU not supported yet")
-            raise RuntimeError(
-                "Error initializing compute.position_lattice_field")
 
-        self.compute_name = "lattice_field"
-        enlist = hoomd.hpmc.data._param.ensure_list
-        self.cpp_compute = cls(hoomd.context.current.system_definition,
-                               enlist(position), float(k), enlist(orientation),
-                               float(q), enlist(symmetry))
-        hoomd.context.current.system.addCompute(self.cpp_compute,
-                                                self.compute_name)
-        if not composite:
-            mc.set_external(self)
+class Harmonic(ExternalField):
+    r"""Restrain particle positions and orientations with harmonic springs.  # noqa: E501,D415,W505
 
-    def set_references(self, position=[], orientation=[]):
-        R""" Reset the reference positions or reference orientations.
+    Args:
+        reference_positions ((*N_particles*, 3) `numpy.ndarray` of
+            `float`): the reference positions, to which particles are restrained
+            :math:`[\mathrm{length}]`.
+        reference_orientations ((*N_particles*, 4) `numpy.ndarray` of
+            `float`): the reference orientations, to which particles are
+            restrained :math:`[\mathrm{dimensionless}]`.
+        k_translational (`float`): translational spring constant
+            :math:`[\mathrm{energy} \cdot \mathrm{length}^{-2}]`.
+        k_rotational (`float`): rotational spring constant
+            :math:`[\mathrm{energy}]`.
+        symmetries ((*N_sym*, 4) `numpy.ndarray` of
+            `float`): the orientations that are equivalent through symmetry,
+            i.e., the rotation quaternions that leave the particles unchanged
+            :math:`[\mathrm{dimensionless}]`.
```

Comment:
Add note here that this must include the identity quaternion rather than a note below.